code/modules/antagonists/traitor/objectives/sabotage_machinery.dm

```diff
@@ -122,17 +122,18 @@ GLOBAL_DATUM_INIT(objective_machine_handler, /datum/objective_target_machine_han
 
 /datum/traitor_objective/sabotage_machinery/trap/generate_ui_buttons(mob/user)
 	var/list/buttons = list()
-	if(!tool)
+	if(!traitor_trapper_given)
 		buttons += add_ui_button("", "Pressing this will materialize an explosive trap in your hand, which you can conceal within the target machine", "wifi", "summon_gear")
 	return buttons
 
 /datum/traitor_objective/sabotage_machinery/trap/ui_perform_action(mob/living/user, action)
 	. = ..()
 	switch(action)
 		if("summon_gear")
-			if(tool)
+			if(traitor_trapper_given)
 				return
-			tool = new(user.drop_location())
+			traitor_trapper_given = TRUE
+			var/obj/item/traitor_machine_trapper/tool = new(user.drop_location())
```

Comment:
if you're going to do this, could you actually add the component `/datum/component/traitor_objective/register` like so?
```suggestion
			var/obj/item/traitor_machine_trapper/tool = new(user.drop_location())
			AddComponent(/datum/component/traitor_objective_register, tool, fail_signals = list(COMSIG_PARENT_QDELETING))
```

Reply:
actually nvm this isn't necessary since you can still succeed at the objective